fvm/src/machine/default.rs

```diff
@@ -117,12 +120,24 @@ where
         #[cfg(not(any(test, feature = "testing")))]
         engine.preload(state_tree.store(), builtin_actors.left_values())?;
 
+        let randomness = externs.get_chain_randomness(
+            0,
+            context.epoch,
+            context.initial_state_root.to_bytes().as_slice(),
+        )?;
+
         Ok(DefaultMachine {
             context: context.clone(),
             engine: engine.clone(),
             externs,
             state_tree,
             builtin_actors,
+            id: format_args!(
+                "{}-{}",
+                context.epoch,
+                cid::multibase::encode(cid::multibase::Base::Base58Btc, &randomness[..16])
```

Comment:
Base58 doesn't work on windows (case sensitive). We don't _really_ support it, but I'd just use base32 unless we have a strong reason to use base58.